web/src/components/netflow-traffic.tsx

```diff
@@ -459,41 +514,50 @@ export const NetflowTraffic: React.FC<{
     (fq: FlowQuery, droppedType: MetricType | undefined) => {
       setFlows([]);
 
-      const { getTopologyMetrics } = getFetchFunctions();
+      const selectedPanels = getSelectedPanels();
+      const { getMetrics } = getFetchFunctions();
 
-      const promises: Promise<Stats>[] = [
-        //get bytes or packets
-        getTopologyMetrics(fq, range).then(res => {
-          setMetrics(res.metrics);
-          return res.stats;
-        }),
+      const promises: Promise<Stats>[] = [];
 
-        //run same query on app scope for total flows
-        getTopologyMetrics({ ...fq, aggregateBy: 'app' }, range).then(res => {
-          setTotalMetric(res.metrics[0]);
-          return res.stats;
-        })
-      ];
+      if (!selectedPanels.some(p => [DROPPED_ID_MATCHER, DNS_ID_MATCHER, RTT_ID_MATCHER].includes(p.id))) {
```

Comment:
not sure to understand this logic : if there's any of [drops/dns/rtt] in panels then we don't fetch the bytes/packets metrics? Why?

Reply:
If there is any panel which is not requiring special query (drop / dns / rtt),
then we query the defaults metrics..

An alternative could be to prepend a matcher to these defaults to manage this case the same way as nexts ones but users will loose their previous selections since it's stored in local storage.

Reply:
I might be wrong but I think that `selectedPanels.some(p => [DROPPED_ID_MATCHER, DNS_ID_MATCHER, RTT_ID_MATCHER].includes(p.id))` is always `false`.
This comes down to writing for instance `["drops", "dns", "rtt"].includes("top_avg_rtt_line")`, that's never going to happen (?)
Should it be instead:

```
selectedPanels.some(p => [DROPPED_ID_MATCHER, DNS_ID_MATCHER, RTT_ID_MATCHER].some(matcher => p.id.includes(matcher)))
```
?

Reply:
I think this would be more correct:
```js
      if (selectedPanels.some(p => !p.id.includes(DROPPED_ID_MATCHER) && !p.id.includes(DNS_ID_MATCHER) && !p.id.includes(RTT_ID_MATCHER))) {

      }

```

Reply:
You are right, thanks for catching that !

I applied your last suggestion: https://github.com/netobserv/network-observability-console-plugin/pull/389/commits/15babbec753bad0d12a94a76f6aadda227495c46